applications/PoCS.md

```diff
@@ -0,0 +1,209 @@
+
+#### Substrate implementation Design
+
+To implement PoCS on substrate we will be modifying `pallet-contracts` and  `pallet-staking` under APACHE 2.0 license. We decided to proceed with this approach since both of these pallets provide the primary required functionalities. In addition to this substrate also provides a elaborate user interface for integrating contracts which will ease out our implementation. Since we will be implementing PoCS as a staking mechanism, we will be using `pallet-staking` with BABE + GRANDPA protocols to modify proof of stake. 
+
+#### Outline of Core functionality 
+
+1. **Fields to be added to Pallet-contracts**
+Every newly deployed contract will have three extra storage fields: 
+* `AccountStakeInfo` (struct) : Struct maintainted for each contract call. It is updated everytime a contract is called. 
+    * `owner` (AccountId) - Owner of the contract
+    * `delegateTo` (AccountId) - Set by the deployer and for the consensus to know who is the deployer.  Only deployer can change this field
+    * `delegateAt` (Blocknumber) - The field is the current block height when the delegateTo field is updated
+* `ContractScarcityInfo` (struct field) -  This field will be used to calculate the PoCS stake score. It will be updated anytime a transaction executes which corresponds to that contract. To calculate this we will use 
+    * `reputation` (u64) - a parameter we assign to calculate the reputation of contract which depends on number of times it is called by a user or any other contract  
+    * `recent_blockheight` (BlockNumberFor) - This is introduced to prevent vulnerabilities like DDoS attack
+
+
+2. Staking mechanism: <br>
+In the context of Substrate, the integration of Babe and Grandpa protocols, alongside the Staking Pallet, complements the PoCS mechanism 
+*  Babe's deterministic block production process aligns with PoCS's commitment to security. Validators with higher reputation and weight history, as determined by the scarcity mapping, are incentivized to actively participate in proposing and validating blocks
+* Staking Pallet (Validator Selection): The Staking Pallet integrates with PoCS as validators selected based on their scarcity scores, are entrusted with the responsibility of proposing and validating blocks. This delegation ensures that validators with a proven track record of actively participating in the network are granted the authority to contribute to the consensus process.
+
+
+<b> Protocol design </b> <br>
+<p align="center">
+  <img src="https://jobyreuben.in/assets/pocs/pocs.jpeg">
+</p>
+
+#### What your project is not or will not provide or implement
+We have structured our implementation into 3 milestones. In this grant our focus is to develop a first version which would implement pallet-contract to calculate the  staking score in simplistic yet efficient way with above mentioned fields. It would handle 
```

Comment:
```suggestion
We have structured our implementation into 4 milestones. In this grant our focus is to develop a first version which would implement pallet-contract to calculate the  staking score in simplistic yet efficient way with above mentioned fields. It would handle 
```